### DISCUSSION TITLE
Unable to exclude issues/prs by ID or "key" in GitHub's GraphQL API

### DISCUSSION THREAD

**Discussion by cweckesser** [community/community#38771] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I'm trying to exclude issues and pull requests by ID and "key". By key I mean the unique identifier for issues and pull requests, composed by the number, repository (`repository.name`) and owner (`repository.owner.login`) properties, all available for both issue and pull request types.

I've tried doing this by using the `NOT` or `-` syntax, as explained in the [search syntax article](https://docs.github.com/en/search-github/getting-started-with-searching-on-github/understanding-the-search-syntax#exclude-certain-results) but with no luck.

- Query excluding by ID:

```graphql
query searchIssues() {
  issues: search(query: "org:MY_ORGANIZATION type:issue -id:MY_ISSUE_ID", type: ISSUE, first: 10) {
    issueCount
    nodes {
      ... on Issue {
        id
        number
        title
        state
        body
        repository {
          name
          owner {
            login
          }
        }
      }
    }
  }
}
```
- Query excluding by (partial) "key":

```graphql
query searchIssues() {
  issues: search(query: "org:MY_ORGANIZATION type:issue -number:MY_ISSUE_NUMBER", type: ISSUE, first: 10) {
    issueCount
    nodes {
      ... on Issue {
        id
        number
        title
        state
        body
        repository {
          name
          owner {
            login
          }
        }
      }
    }
  }
}
```
Note: The filtering by "key" query is left incomplete, since just trying to exclude issues by `number` does not work.

**The above queries do not yield any results. It looks like adding the `-id:MY_ISSUE_ID` and `-number:MY_ISSUE_NUMBER` is the main culprit here.**

---

Questions:

- Is there any way filtering by ID is possible?
- Is filtering by `number`, `repository.name` name and `repository.owner.login` even possible?

Thank you very much in advance!

**Comment by LangLangBart** [comment, marked as answer]
What if you try it with `-created: ...` ?

```graphql
{
  search(query: "repo:cli/cli is:issue", type: ISSUE, first: 2) {
    issueCount
    nodes {
      ... on Issue {
        number
        createdAt
        title
        state
      }
    }
  }
}
```

```json
{
  "data": {
    "search": {
      "issueCount": 3111,
      "nodes": [
        {
          "number": 6588,
          "createdAt": "2022-11-09T19:10:39Z",
          "title": "admin-azazel",
          "state": "CLOSED"
        },
        {
          "number": 6586,
          "createdAt": "2022-11-09T02:18:47Z",
          "title": "ok",
          "state": "CLOSED"
        }
      ]
    }
  }
}
```

### Exclude `2022-11-09T19:10:39Z`

```graphql
{
  search(query: "repo:cli/cli is:issue -created:2022-11-09T19:10:39Z", type: ISSUE, first: 2) {
    issueCount
    nodes {
      ... on Issue {
        number
        createdAt
        title
        state
      }
    }
  }
}
```

```json
{
  "data": {
    "search": {
      "issueCount": 3110,
      "nodes": [
        {
          "number": 6586,
          "createdAt": "2022-11-09T02:18:47Z",
          "title": "ok",
          "state": "CLOSED"
        },
        {
          "number": 6582,
          "createdAt": "2022-11-08T14:13:46Z",
          "title": "Add `--watch` option to `gh pr status` ",
          "state": "OPEN"
        }
      ]
    }
  }
}
```

---

You are limited to these qualifiers: [Searching issues and pull requests](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests)

**Reply by cweckesser** [reply]
@LangLangBart Thanks for the answer. It's not a very clean approach, but I see how it could work (chances of having more than one issue or PR created in the exact same timestamp are extremely low).

**Reply by LangLangBart** [reply]
I agree, it's a foul solution. Maybe the developers are watching in the background and we get an `ID` qualifier. :pray:
